x-team/xp-ui
# Create Timeframe component

**Release Type:** *Non-breaking changes*

Fixes https://x-team-internal.atlassian.net/browse/XP-3701

## Description

Create Timeframe component.

## Checklist


## Steps to Test or Reproduce

- Check the showcase in Storybook http://localhost:9001/?path=/story/core-components-form-components-timeframe--basic-usage

Please take a look at the other Debug showcases, it has some edge cases.

For instance, the start date should never be bigger than an end date. So: 

- If the **start year** is the same of the **end year**, then only the available months to select in the **end month** will be bigger or equal than the **start month**
- If the **start year** is the smaller of the **end year**, all **end month** with be available
- The **end year** options will only have years equal or bigger than **start year**
- If the `Still working there` is checked, then the value returned will ignore the end date (as end date will be undefined)

## Impacted Areas in Application

New `Timeframe` component was added.

## Screenshots

![2020-04-24 12 13 37](https://user-images.githubusercontent.com/131859/80228207-215ee300-8625-11ea-8440-ea40e33ab782.gif)
